packages/core/src/contents/redux/actions/doGetContentPages.js

```diff
@@ -0,0 +1,70 @@
+import * as actionTypes from '../actionTypes';
+import { buildContentGroupHash } from '../../utils';
+import { normalize } from 'normalizr';
+import contentGroup from '../../../entities/schemas/contentGroup';
+
+/**
+ * @enum {ContentType}
+ */
+let ContentType = 'PRODUCT' | 'LISTING' | 'SET';
+
+/**
+ * @typedef {object} GetContentPagesQuery
+ * @property {ContentType} type - Query by a page type (E.g. PRODUCT, LISTING, SET).
+ * @property {string} [slug] - Query by a specified slug.
+ * @property {string} [strategy] - Query by a specific strategy (default).
+ */
+
+/**
+ * @callback GetContentPagesThunkFactory
+ * @param {string} slug - List of codes that representing the content code (about-us|today-news|header|productId...).
+ * @param {GetContentPagesQuery} query - Query object with content pages parameters to apply.
+ * @param {object} [config] - Custom configurations to send to the client
+ * instance (axios).
+ *
+ * @returns {Function} Thunk to be dispatched to the redux store.
+ */
+
+/**
+ * Load a content page ranked for a specific query object received.
+ *
+ * @function doGetContentPages
+ * @memberof module:contents/actions
+ *
+ * @param {Function} getContentPages - Get content pages client.
+ *
+ * @returns {GetContentPagesThunkFactory} Thunk factory.
+ */
+export default getContentPages => (slug, query, config) => async dispatch => {
+  const hash = buildContentGroupHash({
+    contentTypeCode: 'content_pages',
+    codes: slug,
+  });
+
+  dispatch({
+    meta: { query },
+    payload: { hash },
+    type: actionTypes.GET_CONTENT_PAGES_REQUEST,
+  });
+
+  try {
+    const result = await getContentPages(slug, query, config);
+
+    dispatch({
+      meta: { query },
+      payload: {
+        ...normalize({ hash, ...result }, contentGroup),
+        hash,
+      },
+      type: actionTypes.GET_CONTENT_PAGES_SUCCESS,
+    });
```

Comment:
Should we return the result?

Reply:
In the Core, there is no return result. I follow other implementations to be consistent